api/app/src/routes/medical/patient.ts

```diff
@@ -0,0 +1,82 @@
+import { Request, Response } from "express";
+import Router from "express-promise-router";
+import { asyncHandler, getCxIdOrFail, getFacilityIdFromQueryOrFail } from "../util";
+const router = Router();
+import status from "http-status";
+import { getOrganizationOrFail } from "../../command/medical/organization/get-organization";
+import { Patient } from "../../models/medical/patient";
+import { getPatients } from "../../command/medical/patient/get-patient";
+import { updatePatient } from "../../command/medical/patient/update-patient";
+import { createPatient } from "../../command/medical/patient/create-patient";
+
+/** ---------------------------------------------------------------------------
+ * POST /patient
+ *
+ * Updates or creates the patient corresponding to the specified facility at the
+ * customer's organization if it doesn't exist already.
+ *
+ * @return  {Facility}  The facility.
+ */
+router.post(
+  "/",
+  asyncHandler(async (req: Request, res: Response) => {
+    const cxId = getCxIdOrFail(req);
+    const org = await getOrganizationOrFail({ cxId });
+    const facilityId = getFacilityIdFromQueryOrFail(req);
+
+    // TODO: parse this into model
+    const patientData = req.body;
+
+    let patient: Patient;
+    if (patientData.id) {
+      const data = { ...patientData };
+      delete data.id;
+      delete data.organizationId;
+      delete data.facilityId;
+      patient = await updatePatient({
+        id: patientData.id,
+        cxId,
+        organizationId: org.organizationId,
+        facilityId: +facilityId,
+        data,
+      });
+    } else {
+      patient = await createPatient({
+        cxId,
+        data: patientData,
+        organizationId: org.organizationId,
+        facilityId: +facilityId,
+      });
+    }
+
+    // TODO: create or update patient in CW as well
```

Comment:
As the "create or update" logic is <20 LOC, and most likely will only ever be used here, you okay to leave this until we actually use it in another part of the API?

Reply:
We can defs leave here, but its about defining patterns. Others might base off this, and it might not be small.

Yet, I'd argue its enough logic to warrant a command. :)

But cool if you want to push as is for the mvp